### PR TITLE
adds comments endpoint

### PR DESCRIPTION
**Pull request by TravelByBeaut**
Adds statuses 200, 400 and 404

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -159,3 +159,38 @@ describe("Users", () => {
...
+        .then((response) => {
+          expect(response.body.msg).toBe("Bad request");
+        });
+    });
+    test("status: 404 when given an valid but non-existent address", () => {
+      return request(app)
+        .get("/api/articles/100/comments")
+        .expect(404)
+        .then((response) => {
+          expect(response.body.msg).toBe("Invalid address");
```
Comment (Rosem10) [inline __tests__/app.test.js]: I wouldn't call this a non-existent address. The endpoint is valid, we just don't have a corresponding comment with that id.

File: __tests__/app.test.js [hunk 2]
```diff
@@ -159,3 +159,38 @@ describe("Users", () => {
     });
   });
 });
+describe("Comments", () => {
+  describe("getCommentsById", () => {
+    test("status: 200 and resonds with an array of comments objects relating to the article_id", () => {
+      return request(app)
+        .get("/api/articles/1/comments")
+        .expect(200)
+        .then(({ body: { article } }) => {
```
Comment (Rosem10) [inline __tests__/app.test.js]: This should be being sent back on a key of comments please! It's an array of comments, not an array of articles.

File: models/news.model.js [hunk 3]
```diff
@@ -59,3 +59,20 @@ exports.articleDataByDate = () => {
       return article;
     });
 };
+
+exports.commentsById = (id) => {
+  return db
+    .query(
+      `SELECT articles.article_id, comments.* FROM articles
```
Comment (Rosem10) [inline models/news.model.js]: This isn't necessary for commentsById! But it is needed for add comment count to article. We just need to select all from the comments where the article_id matches the one passed. It is also here that we should check the article exists before checking the comments for it.

**Review by Rosem10 — CHANGES_REQUESTED** [review]
Have another look please!

File: models/news.model.js (hunk 3 above)
Comment (TravelByBeaut) [reply models/news.model.js]: I think this checks if it exists before looking for the comments unless there is specific code to be in there. I looked in the notes in complex queries but it is also talking about WHERE .... which i have in the query anyway. Let me know if I need to add something.

Comment (Rosem10) [reply models/news.model.js]: So I see what you're getting at with this, but we need to consider how will it handle the difference between an article that doesn't exist VS an article that exists but has no comments. if the article exists but has no comments, our rows on line 72 will be returned as an empty array. However, this will also be the case if we ask SQL to go in and select any comments where that article_id matches. It won't error out - because we haven't done any bad sql - but nor will the array of rows returned have anything inside it. We have to tell our code how to tell the difference.

**Review by Rosem10 — CHANGES_REQUESTED** [review]

File: __tests__/app.test.js [hunk 4]
```diff
@@ -159,3 +159,37 @@ describe("Users", () => {
...
+          expect(response.body.msg).toBe("Bad request");
+        });
+    });
+    test("status: 404 when given an valid address but no data", () => {
+      return request(app)
+        .get("/api/articles/100/comments")
+        .expect(404)
+        .then((response) => {
+          expect(response.body.msg).toBe("Article_id doesn't exist");
+        });
```
Comment (HeatherEver) [inline __tests__/app.test.js:200]: Can you please add a test for an article with no comments coming back with a status 200 and empty array

**Review by HeatherEver — COMMENTED** [review]
Nice work getting all those comments sorted - looks good, just one test to add!

**Review by HeatherEver — APPROVED** [review]
Nice work! Glad it got sorted 🎉
